### PR TITLE
Refactor Lidar drivers

### PR DESCRIPTION
**Pull request by lx18233184051**
1: keep original component to adapt dreamview.
2: modify all brand lidar(velodyne hesai robosense) driver to adbstact factory , so that developer can use LidarDriverFactory to add new lidar easily.
3: TODO: abstact parse module.
4: this modify has tested on D-kit.

### CODE REVIEW

File: modules/drivers/lidar/common/driver_factory/driver_base.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
```
Comment (changsh726) [inline modules/drivers/lidar/common/driver_factory/driver_base.h:2]: 2020

File: modules/drivers/lidar/velodyne/driver/driver.h [hunk 2]
```diff
@@ -71,10 +82,16 @@ class VelodyneDriver {
 class Velodyne64Driver : public VelodyneDriver {
  public:
   explicit Velodyne64Driver(const Config &config) : VelodyneDriver(config) {}
-  ~Velodyne64Driver() {}
-
-  void Init() override;
+  explicit Velodyne64Driver(const std::shared_ptr<cyber::Node> &node,
```
Comment (changsh726) [inline modules/drivers/lidar/velodyne/driver/driver.h:85]: no need explicit here

File: modules/drivers/lidar/velodyne/driver/driver.h [hunk 3]
```diff
@@ -71,10 +82,16 @@ class VelodyneDriver {
...
+                            const Config &config) {
+    node_ = node;
+    config_ = config;
+  }
+  ~Velodyne64Driver();
+
+  bool Init() override;
   bool Poll(const std::shared_ptr<VelodyneScan> &scan) override;
+  void device_poll();
```
Comment (changsh726) [inline modules/drivers/lidar/velodyne/driver/driver.h:94]: DevicePoll()

File: modules/drivers/lidar/velodyne/driver/driver.h [hunk 4]
```diff
@@ -38,18 +41,27 @@ constexpr double PACKET_RATE_HDL64E_S3D = 5789;
 constexpr double PACKET_RATE_VLS128 = 6250.0;
 constexpr double PACKET_RATE_VLP32C = 1507.0;
 
-class VelodyneDriver {
+class VelodyneDriver : public lidar::LidarDriver {
  public:
+  VelodyneDriver() {}
   explicit VelodyneDriver(const Config &config) : config_(config) {}
+  explicit VelodyneDriver(const std::shared_ptr<cyber::Node> &node,
```
Comment (changsh726) [inline modules/drivers/lidar/velodyne/driver/driver.h:48]: no need explicit here

File: modules/drivers/lidar/common/driver_factory/BUILD [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+cc_library(
+    name = "lidar_driver_factory",
+    srcs = ["lidar_driver_factory.cc"],
+    hdrs = ["lidar_driver_factory.h"],
+    deps = [
+        "//cyber",
+        "//modules/common/util:factory",
+        ":driver_base",
+        "//modules/drivers/lidar/proto:lidar_parameter",
```
Comment (storypku) [inline modules/drivers/lidar/common/driver_factory/BUILD:14]: Please run buildifer or apollo.sh format on this file.

File: modules/drivers/lidar/common/driver_factory/driver_base.h [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+ * @class LidarDriver
+ * @brief The class which defines the lidar driver .
+ */
+class LidarDriver {
+ public:
+  /**
+   * @brief Constructor
+   */
+  LidarDriver() {}
+  explicit LidarDriver(const std::shared_ptr<::apollo::cyber::Node>& node) {}
```
Comment (storypku) [inline modules/drivers/lidar/common/driver_factory/driver_base.h:51]: node_ should be initialized with node.

File: modules/drivers/lidar/common/driver_factory/driver_base.h [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~LidarDriver() = default;
+
+  /**
+   * @brief Initialize the CAN client by specified CAN card parameters.
+   * @param parameter CAN card parameters to initialize the CAN client.
+   * @return If the initialization is successful.
```
Comment (storypku) [inline modules/drivers/lidar/common/driver_factory/driver_base.h:61]: Comment is misleading.

**Review by storypku — CHANGES_REQUESTED** [review]

**Review by lfcarol — DISMISSED** [review]

**Review by lfcarol — APPROVED** [review]

**Pull request merged**